src/test/java/teammates/test/cases/ui/pagedata/StudentCommentsPageDataTest.java

```diff
@@ -0,0 +1,231 @@
+        FeedbackResponseComment expectedFeedbackResponseCommentRow = 
+                new FeedbackResponseComment(responseComment, expectedGiverDetails);
+        
+        String giverName = student1.name + " (" + student1.team + ")";
+        String recipientName = giverName;
+        String feedbackResponse = response.getResponseDetails().getAnswerHtml(question.getQuestionDetails());
+        
+        ResponseRow expectedResponseRow = 
+                new ResponseRow(giverName, recipientName, feedbackResponse, 
+                                Arrays.asList(expectedFeedbackResponseCommentRow));
+        
+        QuestionTable expectedQuestionTable = 
+                new QuestionTable(question.questionNumber, 
+                                  results.getQuestionText(question.getId()), 
+                                  question.getQuestionDetails()
+                                                   .getQuestionAdditionalInfoHtml(question.questionNumber, ""), 
+                                  Arrays.asList(expectedResponseRow));
+        FeedbackSessionRow expectedFeedbackSessionRow = 
+                new FeedbackSessionRow(
+                        session.feedbackSessionName, session.courseId, Arrays.asList(expectedQuestionTable));
+        assertEquals(1, actualFeedbackSessionRows.size());
+
+        FeedbackSessionRow actualFeedbackSessionRow = actualFeedbackSessionRows.get(0);
+        assertEquals(expectedFeedbackSessionRow.getFeedbackSessionName(), 
+                     actualFeedbackSessionRow.getFeedbackSessionName());
+        assertEquals(expectedFeedbackSessionRow.getCourseId(), actualFeedbackSessionRow.getCourseId());
+        assertEquals(1, actualFeedbackSessionRow.getQuestionTables().size());
+        
+        QuestionTable actualQuestionTable = actualFeedbackSessionRow.getQuestionTables().get(0);
+        assertEquals(expectedQuestionTable.getQuestionNumber(), actualQuestionTable.getQuestionNumber());
+        assertEquals(expectedQuestionTable.getQuestionText(), actualQuestionTable.getQuestionText());
+        assertEquals(expectedQuestionTable.getAdditionalInfo(), actualQuestionTable.getAdditionalInfo());
+        assertEquals(1, actualQuestionTable.getResponseRows().size());
+        
+        ResponseRow actualResponseRow = actualQuestionTable.getResponseRows().get(0);
+        assertEquals(expectedResponseRow.getGiverName(), actualResponseRow.getGiverName());
+        assertEquals(expectedResponseRow.getRecipientName(), actualResponseRow.getRecipientName());
+        assertEquals(expectedResponseRow.getResponse(), actualResponseRow.getResponse());
+        assertEquals(1, actualResponseRow.getFeedbackResponseComments().size());
+        
+        FeedbackResponseComment actualFeedbackResponseCommentRow = 
+                actualResponseRow.getFeedbackResponseComments().get(0);
+        checkFeedbackResponseCommentRowsEqual(expectedFeedbackResponseCommentRow, actualFeedbackResponseCommentRow);
```

Comment:
This test method is quite long. See if you can shorten it using some more SLAP. i.e. extract lower level details to methods so that the higher level logic becomes easier to see.